test/ZNSRootRegistrar.test.ts

```diff
@@ -103,6 +103,73 @@ describe("ZNSRootRegistrar", () => {
     await mongoAdapter.dropDB();
   });
 
+  it("Should register an array of domains", async () => {
+    const registrations = [];
+
+    for (let i = 0; i < 5; i++) {
+      const isOdd = i % 2 !== 0;
+
+      const domainObj = {
+        name: `domain${i + 1}`,
+        domainAddress: user.address,
+        tokenURI: `0://domainURI_${i + 1}`,
+        distributionConfig: {
+          pricerContract: await zns.curvePricer.getAddress(),
+          paymentType: isOdd ? PaymentType.STAKE : PaymentType.DIRECT,
+          accessType: isOdd ? AccessType.LOCKED : AccessType.OPEN,
+        },
+        paymentConfig: {
+          token: await zns.meowToken.getAddress(),
+          beneficiary: isOdd ? user.address : operator.address,
+        },
+      };
+
+      registrations.push(domainObj);
+    }
+
+    await zns.rootRegistrar.connect(user).registerMultipleRootDomains(registrations);
+
+    for (const domain of registrations) {
+      // get by `domainHash`
+      const logs = await getDomainRegisteredEvents({
+        zns,
+        domainHash: hashDomainLabel(domain.name),
+      });
+
+      // "DomainRegistered" event log
+      const args = logs[0].args;
+
+      expect(args[0]).to.eq(ethers.ZeroHash); // parentHash
+      expect(args[1]).to.eq(hashDomainLabel(domain.name)); // domainHash
+      expect(args[2]).to.eq(domain.name); // label
+      expect(args[4]).to.eq(domain.tokenURI); // tokenURI
+      expect(args[5]).to.eq(user.address); // registrant
+      expect(args[6]).to.eq(domain.domainAddress); // domainAddress
+    }
+  });
+
+  it("Should revert when register the same domain twice using #registerMultipleRootDomains", async () => {
+    const domainObj = {

```

Comment:
same as above, create and use a type if one doesnt already exist. This will make sure that all these tests use the same thing and future proof it